ImperatorToCk2/src/main/java/ImperatorToCK2/Processing.java

```diff
@@ -1039,4 +1038,20 @@ else if (provName.equals("kyzyl_su")) {
         return cultureTitles;
     }
 
+    public static String checkFile(String directory) throws IOException //checks if file exists or not
+    {
+
+        try {
+
+            FileInputStream fileIn= new FileInputStream(directory);
+            Scanner scnr= new Scanner(fileIn);
+
+        }catch (java.io.FileNotFoundException exception){
+            return "no";
+
+        }
+
+        return "yes";
+    }
```

Comment:
The entire function is too complex. There's no need for doing something that might throw an exception (exceptions should be for exceptional circumstances, not something normal). Instead, just do

```
File f = new File(directory);
if(f.exists() && !f.isDirectory()) { 
    return true;
}

return false;
```